applications/StatisticsApplication/python_scripts/temporal_statistics_process.py

```diff
@@ -57,6 +57,11 @@ def __init__(self, model, settings):
         if (not Kratos.KratosGlobals.HasVariable(statistics_control_variable_name)):
             raise Exception("Unknown statistics control variable. [ \"statistics_control_variable_name\" = \"" + statistics_control_variable_name + "\" ]")
 
+        ## this is required to support restarting capabilities. If STEP is used, there need to be a way to retrieve
```

Comment:
Temporal methods do time integration, if STEP to be used as integration controlling variable, then I would have to find out TIME corresponding to STEP which integration started and record it somewhere for each temporal method to be reused in restart. (Eventhough automatic time stepping is not that promoted, I kept in mind automatic time stepping as well for temporal time integration methods, therefore having a record of time integration starting TIME of the STEP is crucial). So If I'm to use STEP, I end up having a local runtime variable per temporal method of integration which I need to store in restart files. Since these temporal methods are objects which lies in python level, not in model part, I didnt find a cleaner solution to integrate them to model part than hackish ways. So I left it out having this comment so someday, if someone or I find a way to do it cleaner then it can be implemented.